libs/feature-recipe-details/src/lib/feature-recipe-details/feature-recipe-details.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable, inject } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute, Router } from '@angular/router';
+import {
+  RecipeDto,
+  RecipesApiActions,
+  RecipesUiActions,
+  getSelectedRecipe,
+} from '@hoa-recipes/data-access-recipes';
+import { Actions, ofType } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { distinctUntilChanged } from 'rxjs';
+
+@Injectable()
+export class FeatureRecipeDetailsService {
+  private store = inject(Store);
+  private route = inject(ActivatedRoute);
+  private actions = inject(Actions);
+  private dialog = inject(MatDialog);
+  private router = inject(Router);
+
+  recipe = this.store.selectSignal(getSelectedRecipe);
+
+  constructor() {
+    this.route.paramMap
+      .pipe(distinctUntilChanged(), takeUntilDestroyed())
+      .subscribe((r) => {
+        const recipeId = r.get('id');
+        this.store.dispatch(
+          RecipesUiActions.selectRecipe({ selectedId: recipeId as string }),
+        );
+      });
+
+    this.actions
+      .pipe(ofType(RecipesApiActions.deleteRecipeSuccess), takeUntilDestroyed())
+      .subscribe(() => {
+        this.router.navigate(['/']);
+      });
+
+    this.actions
+      .pipe(
+        ofType(RecipesApiActions.uploadRecipeImageSuccess),
+        takeUntilDestroyed(),
+      )
+      .subscribe((res) => {
+        this.store.dispatch(
+          RecipesUiActions.editRecipe({
+            recipeDto: {
+              ...(this.recipe() as RecipeDto),
+              imageName: res.fileName,
+            },
+          }),
+        );
+        this.dialog.closeAll();
+      });
```

Comment:
this feels wrong; looks like uploading an image submits a form (?)

Reply:
uploading an image is a local rather than global state. There's no need to go through ngrx as it's coupled to the form.

Reply:
Yes, I did this on purpose. The problem is that the user can upload any format of image (png, jpg, jpeg, webp, etc.). So what I do is the following
1. When user click on image upload button
2. Check image extension and upload it to the backend
3. On successful response update the `imageName` property on the recipe - hence the form submission

On a 'normal' backend I would handle this with updateRecipe function. But because of the limitations of json-server and/or multer I did it this way.

Let me know if I should really fix this. This was kind of a practice to me, to see what ngrx is capable of. 

**Question**: Is it common practice to chain Ngrx actions/effects to achieve similar goals? Like when I need to update more than one resource on the backend?